### PR TITLE
Create image and transformation IO page

### PR DESCRIPTION
**Pull request by seungkyoon**
Describes supported transformation and image files and includes
simple examples of how to read and write both. Also includes reference
in FAQ and TOC

### CODE REVIEW

File: Documentation/docs/source/IO.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
+Image and Transformation IO
+***************************
+
+.. _image-io:
+
+Images
+======
+
+The default SimpleITK ``ImageFileReader`` can handle numerous file types using
```
Comment (blowekamp) [inline Documentation/docs/source/IO.rst]: We should introduce what an ImageIO is little, consider working in something like the following:

There are numerous file formats support by SimpleITK's image readers and writers. Support for a particular format is handled by a specific ITK `ImageIO`(link) class.  By default, the ImageIO is automatically determined for a particular file. Advanced SimpleITK installations can configure or extend which file formats are supported by SimpleITK.

File: Documentation/docs/source/IO.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        writer.SetFileName(outputImageFileName)
+        writer.Execute(image)
+
+
+.. _transformation-io:
+
+Transformations
+===============
+
+In SimpleITK, transformation files are written as .tfm files. Take an example of a
```
Comment (blowekamp) [inline Documentation/docs/source/IO.rst]: There are other extensions supported, by the conventional ITK text file format:
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/IO/TransformInsightLegacy/include/itkTxtTransformIO.hxx#L61

There is also the the MINCTransformIO, TransformHDF5, TransformMatlab, TransformMINC transform IO.
https://github.com/InsightSoftwareConsortium/ITK/tree/master/Modules/IO

File: Documentation/docs/source/IO.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        sitk.WriteTransform(basic_transform, full_file_name)
+        read_result = sitk.ReadTransform(full_file_name)
+
+        assert(str(type(read_result) != type(basic_transform)))
+
+``read_result`` will be an object of the generic ``sitk.Transform()`` class and ``basic_transform``
+will be of ``sitk.Euler2DTransform()``, but both represent the same transformation. The last line
+asserts that the two are of different types.
+
+*euler2D.tfm* contains:
```
Comment (blowekamp) [inline Documentation/docs/source/IO.rst]: We don't need the file example here.

File: Documentation/docs/source/IO.rst [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+.. code-block :: python
+
+        import os
+        import SimpleITK as sitk
+        import numpy as np
+
+        basic_transform = sitk.Euler2DTransform()
+        basic_transform.SetTranslation((1,2))
+        basic_transform.SetAngle(np.pi/2)
```
Comment (blowekamp) [inline Documentation/docs/source/IO.rst]: make the sample shorter, skip the imports, np stuff, path details, focus on just the SimpleITK code.

**Comment by seungkyoon** [comment]
Fixed changes

**Comment by blowekamp** [comment]
Great! Merging to next.

**Pull request merged**